web_analytics.go

```diff
@@ -0,0 +1,393 @@
+	Inclusive bool      `json:"inclusive"`
+	Created   time.Time `json:"created,omitempty"`
+	// IsPaused defines whether the rule is paused (inactive) or not.
+	IsPaused bool `json:"is_paused"`
+	Priority int  `json:"priority,omitempty"`
+}
+
+// CreateWebAnalyticsRule describes the properties required to create or update a Web Analytics Rule object.
+type CreateWebAnalyticsRule struct {
+	ID    string   `json:"id,omitempty"`
+	Host  string   `json:"host"`
+	Paths []string `json:"paths"`
+	// Inclusive defines whether the rule includes or excludes the matched traffic from being measured in web analytics.
+	Inclusive bool `json:"inclusive"`
+	IsPaused  bool `json:"is_paused"`
+}
+
+// WebAnalyticsRuleset describes a Web Analytics Ruleset object.
+type WebAnalyticsRuleset struct {
+	ID       string `json:"id"`
+	ZoneTag  string `json:"zone_tag"`
+	ZoneName string `json:"zone_name"`
+	Enabled  bool   `json:"enabled"`
+}
+
+// WebAnalyticsSiteResponse is the API response, containing a single WebAnalyticsSite.
+type WebAnalyticsSiteResponse struct {
+	Response
+	Result WebAnalyticsSite `json:"result"`
+}
+
+// WebAnalyticsSitesResponse is the API response, containing an array of WebAnalyticsSite.
+type WebAnalyticsSitesResponse struct {
+	Response
+	ResultInfo ResultInfo         `json:"result_info"`
+	Result     []WebAnalyticsSite `json:"result"`
+}
+
+// WebAnalyticsRuleResponse is the API response, containing a single WebAnalyticsRule.
+type WebAnalyticsRuleResponse struct {
+	Response
+	Result WebAnalyticsRule `json:"result"`
+}
+
+type WebAnalyticsRulesetRules struct {
+	Ruleset WebAnalyticsRuleset `json:"ruleset"`
+	Rules   []WebAnalyticsRule  `json:"rules"`
+}
+
+// WebAnalyticsRulesResponse is the API response, containing a WebAnalyticsRuleset and array of WebAnalyticsRule.
+type WebAnalyticsRulesResponse struct {
+	Response
+	Result WebAnalyticsRulesetRules `json:"result"`
+}
+
+// WebAnalyticsIDResponse is the API response, containing a single ID.
+type WebAnalyticsIDResponse struct {
+	Response
+	Result struct {
+		ID string `json:"id"`
+	} `json:"result"`
+}
+
+// WebAnalyticsSiteTagResponse is the API response, containing a single ID.
+type WebAnalyticsSiteTagResponse struct {
+	Response
+	Result struct {
+		SiteTag string `json:"site_tag"`
+	} `json:"result"`
+}
+
+type CreateWebAnalyticsSiteParams struct {
+	// Host is the host to measure traffic for.
+	Host string `json:"host,omitempty"`
+	// ZoneTag is the zone tag to measure traffic for.
+	ZoneTag string `json:"zone_tag,omitempty"`
+	// AutoInstall defines whether Cloudflare will inject the JS snippet automatically for orange-clouded sites.
+	AutoInstall bool `json:"auto_install"`
+}
+
+// CreateWebAnalyticsSite creates a new Web Analytics Site for an Account.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-create-site
+func (api *API) CreateWebAnalyticsSite(ctx context.Context, rc *ResourceContainer, params CreateWebAnalyticsSiteParams) (*WebAnalyticsSite, error) {
```

Comment:
if this is an account only route, we should add some validation like https://github.com/cloudflare/cloudflare-go/blob/92917bc2951b51f6a616c405e7aaa0aca8ada635/workers_bindings.go#L437-L439 to prevent attempting to call this with a zone identifier.